utils/redis_manager.py

```
@@ -0,0 +1,293 @@
+            value = args[1] if len(args) > 1 else kwargs.get('value')
+            self._fallback_storage[key] = value
+            return True
+        elif operation == 'delete':
+            key = args[0] if args else kwargs.get('key')
+            return self._fallback_storage.pop(key, None) is not None
+        elif operation == 'exists':
+            key = args[0] if args else kwargs.get('key')
+            return key in self._fallback_storage
+        elif operation == 'clear_pattern':
+            pattern = args[0] if args else kwargs.get('pattern')
+            keys_to_remove = [k for k in self._fallback_storage.keys() if pattern in k]
```

Ruff (0.12.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)